### PR TITLE
Add net_iface config option.

### PR DESCRIPTION
**Pull request by ratmole**
i needed this!
seems to work :)

### CODE REVIEW

File: bpytop.py [hunk 1]
```diff
@@ -3193,6 +3198,9 @@ def _get_nics(cls):
 			cls.nics.append(nic)
 		if not cls.nics: cls.nics = [""]
 		cls.nic = cls.nics[cls.nic_i]
+		if cls.net_iface:
```
Comment (aristocratos) [inline bpytop.py:3201]: Should probably add a condition to check that `net_iface` is a correct nic name. Something like:
`if cls.net_iface and cls.net_iface in cls.nics:`

**Comment by aristocratos** [comment]
@ratmole 
Looks good. Will merge if you add the additional condition to check that the nic is valid.

File: bpytop.py [hunk 2]
```diff
@@ -3193,6 +3198,9 @@ def _get_nics(cls):
 			cls.nics.append(nic)
 		if not cls.nics: cls.nics = [""]
 		cls.nic = cls.nics[cls.nic_i]
+		if cls.net_iface: 
+                        cls.nic = cls.net_iface
```
Comment (UmarJ) [inline bpytop.py:3202]: Should add `cls.nic_i = cls.nics.index(cls.nic)` here, for smooth switching behaviour.

**Comment by UmarJ** [comment]
@ratmole 
Oh, and also set `cls.net_iface` to `""` if the nic is switched, or it's going to switch to that nic every time the nic list is updated.

**Comment by aristocratos** [comment]
@UmarJ
Good catch.

**Comment by ratmole** [comment]
> @ratmole
> Oh, and also set `cls.net_iface` to `""` if the nic is switched, or it's going to switch to that nic every time the nic list is updated.

not following :(
can you please give me a hint?

**Comment by aristocratos** [comment]
@ratmole 
> can you please give me a hint?

`cls.net_iface` needs to be cleared after setting the nic at line 3204, otherwise the current nic will automatically change to the `cls.net_iface` anytime the nic list is updated regardless of which nic you are currently viewing.

**Comment by UmarJ** [comment]
@ratmole, At the start of this function, add `if cls.net_iface: cls.net_iface = ""`

https://github.com/aristocratos/bpytop/blob/f0cbac63e4259ad2955cd4ceb6f9a8b84c050620/bpytop.py#L3206-L3219

**Comment by aristocratos** [comment]
@ratmole 
Looks good, will be merged in next update.
Thank you for contributing.

**Pull request merged**